scripts/prepare_vehicle_charging_demand.py

```diff
@@ -0,0 +1,189 @@
+    "work" and "home" are smoothed between `HOME_START`, `HOME_END` and `WORK_START`, `WORK_END`
+    respectively with charging strategy "balanced".
+    The charging demand profile is then divided by the total demand.
+
+    Parameters
+    ----------
+    input_dir : str
+        Path to directory where csv files containing electric vehicle charging demand profiles
+        are placed
+    balanced : bool
+        If True profiles "work" and "home" are smoothed with charging strategy "balanced".
+        Default: True
+
+    Returns
+    -------
+    ts_prepared : pd.DataFrame
+        Contains electric vehicle charging demand profiles in the format of time series template of
+        oemof-B3
+
+    """
+    # initialize data frame
+    df = pd.DataFrame()
+
+    # loop through files in `input_dir`
+    for filename in os.listdir(input_dir):
+        # get year and region from file name
+        split_filename = os.path.splitext(os.path.basename(filename))[0].split("_")
+        region, year = split_filename[2], int(split_filename[3])
+        path = os.path.join(input_dir, filename)
+
+        # read data from file, copy and superfluous drop last time step
+        ts_raw = pd.read_csv(
+            path,
+            index_col=1,
+            sep=";",
+            decimal=",",
+            parse_dates=True,
+            thousands=".",
+        ).drop(columns=["Unnamed: 0"], axis=1)
+        ts = ts_raw[ts_raw.index.year == year]
+
+        # resample (15 min to hourly), unit is kW
+        hourly_ts = ts.resample("H").mean()
+
+        if balanced:
+            # smooth work and home profiles as they have high peaks (~strategy balanced)
+            hourly_ts = balance_profiles(df=hourly_ts)
+
+        # only keep column "sum CS power" (sum of power demand at all charging stations)
+        ts_total_demand = pd.DataFrame(hourly_ts["sum CS power"]).rename(
+            columns={"sum CS power": f"ts_{year}"}
+        )
+
+        # divide by total electricity demand of vehicles
+        ts_total_norm = ts_total_demand / ts_total_demand.sum()
+
+        # stack time series and add region
+        ts_stacked = dp.stack_timeseries(ts_total_norm).rename(
+            columns={"var_name": "scenario"}
+        )
+        ts_stacked.loc[:, "region"] = region
+
+        # add to `df`
+        df = pd.concat([df, ts_stacked], axis=0)
+
+    # prepare index
+    df.reset_index(inplace=True, drop=True)
+    df.index.name = "id_ts"
+
+    # bring time series to oemof-B3 format with `stack_timeseries()` and `format_header()`
+    ts_prepared = dp.format_header(df=df, header=dp.HEADER_B3_TS, index_name="id_ts")
+
+    # add additional information as required by template
+    ts_prepared.loc[:, "var_unit"] = TS_VAR_UNIT
+    ts_prepared.loc[:, "var_name"] = "vehicle-charging-profile"
+    ts_prepared.loc[:, "source"] = TS_SOURCE
+    ts_prepared.loc[:, "comment"] = TS_COMMENT
+
+    return ts_prepared
+
+
+def balance_profiles(df):
```

Comment:
This function is not generic but created for better overview in `prepare_vehicle_time_series`; better include it as function in function `prepare_vehicle_time_series`?

Reply:
I keep it outside as it is indeed quite big